common/media-type/src/main/java/io/helidon/common/media/type/ParserMode.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2023 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.common.media.type;
+
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * Media type parsing mode.
+ */
+public enum ParserMode {
+
+    /**
+     * Strict mode (default).
+     * Media type must match known name.
+     */
+    STRICT,
+    /**
+     * Relaxed mode.
+     * Apply additional rules to identify unknown media types.
+     */
+    RELAXED;
+
+    // Relaxed media types mapping
+    private static final Map<String, String> RELAXED_TYPES = Map.of(
+            "text", "text/plain"
+    );
+
+    // Lower-case value names mapping
+    private static final Map<String, ParserMode> VALUE_OF_MAPPING = Map.of(
+            STRICT.name().toLowerCase(), STRICT,
+            RELAXED.name().toLowerCase(), RELAXED
+    );
+
+    /**
+     * Find relaxed media type mapping for provided value.
+     *
+     * @param value source media type value
+     * @return mapped media type value or {@code Optional.empty()}
+     *         when no mapping for given value exists
+     */
+    public static Optional<String> findRelaxedMediaType(String value) {
```

Comment:
It should be clear this method is only called when the value is not according to spec.
This actually feels like something we may want to provide full customization for using something like

```
interface MediaTypeParser {
  Optional<MediaType> map(String value);
}
```
That we would implement ourself using what we have in `MediaTypeImpl.parse()`,
and allow customizing it when setting up server or client (similar to the parser mode you have now). This would allow us to create our own parser with the mode (such as `HelidonMediaTypeParser.create()` and `HelidonMediaTypeParser.create(ParserMode)`, or even have it customizable using a builder to provide additional mappings, or to have a falblack type to use when parsing fails.
This seems to be a bit too hard-coded

Reply:
Changed to pkg visibility to remove it from API. Will keep hard-coded version now, but it may be rewritten in a better way.